extensions/ql-vscode/src/pure/location-link-utils.ts

```diff
@@ -13,3 +13,17 @@ export function createRemoteFileRef(
     return `${fileLink.fileLinkPrefix}/${fileLink.filePath}`;
   }
 }
+
+/**
+ * Creates a markdown link to a remote file.
+ * If the "link text" is not provided, we use the file path.
+ */
+export function createMarkdownRemoteFileRef(
+  fileLink: FileLink,
+  startLine?: number,
+  endLine?: number,
+  linkText?: string,
+): string {
+  const markdownLink = `[${linkText || fileLink.filePath}](${createRemoteFileRef(fileLink, startLine, endLine)})`;
```

Comment:
Is `fileLink.filePath` the full path? Maybe we just want the file name without the path?

```suggestion
  const markdownLink = `[${linkText || Path.basename(fileLink.filePath)}](${createRemoteFileRef(fileLink, startLine, endLine)})`;
```

I could go either way with this. It really depends on how and where it is used.

Reply:
Also, even though it _probably_ won't happen, it might be a good idea to check that either both startLine and endLine or neither are specified. If only one is, then we should avoid using it at all.

Reply:
Good point - we check for this in the underlying function [`createRemoteFileRef`](https://github.com/github/vscode-codeql/blob/85e051a76d0b0c072c3c559ef60abe942979df8a/extensions/ql-vscode/src/pure/location-link-utils.ts#L3).